### PR TITLE
Support linux cpp properties

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: resources/templates/esp32/c_cpp_properties_linux.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "configurations": [
+        {
+            "name": "Linux",
+            "defines": [
+                "ARDUINO=10800"
+            ],
+            "includePath": [
+                "${workspaceFolder}",
+                "{ROOTPATH}/.arduino15/packages/AZ3166/tools/**",
```
Comment (dooriya) [inline resources/templates/esp32/c_cpp_properties_linux.json]: Why we need to include "az3166" libraries in ESP32 project?

File: src/Models/ArduinoDeviceBase.ts [hunk 2]
```diff
@@ -191,9 +192,23 @@ export abstract class ArduinoDeviceBase implements Device {
...
+                board.id, constants.cppPropertiesFileNameLinux));
+        const propertiesContentLinux =
+            fs.readFileSync(propertiesFilePathLinux).toString();
+        const rootPathPattern = /{ROOTPATH}/g;
+        const versionPattern = /{VERSION}/g;
+        const homeDir = await IoTWorkbenchSettings.getOs();
+        const replaceStr =
+            propertiesContentLinux.replace(rootPathPattern, homeDir)
+                .replace(versionPattern, this.version);
+        await FileUtility.writeFile(type, cppPropertiesFilePath, replaceStr);
```
Comment (dooriya) [inline src/Models/ArduinoDeviceBase.ts]: Similar code block as "win32" case, suggest moving to a wrapper method to improve readability.

File: src/Models/ArduinoDeviceBase.ts [hunk 3]
```diff
@@ -191,9 +192,23 @@ export abstract class ArduinoDeviceBase implements Device {
...
       }
-      // TODO: Let's use the same file for Linux and MacOS for now. Need to
-      // revisit this part.
+      // TODO: Let's use the MacOS template file for OS system that is not win32
+      // / linux. Revisit this part if want to support other OS system.
```
Comment (dooriya) [inline src/Models/ArduinoDeviceBase.ts]: >OS system [](start = 61, length = 9)

Just OS, no more system

File: resources/templates/esp32/c_cpp_properties_linux.json (hunk 1 above)
Comment (dilin-MS) [reply resources/templates/esp32/c_cpp_properties_linux.json]: It is a bug. Fixed.

File: src/Models/ArduinoDeviceBase.ts (hunk 3 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Fixed.

File: src/Models/ArduinoDeviceBase.ts (hunk 2 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Not all the same since win32 will need changing `\`  to `\\`.

**Pull request merged**